### PR TITLE
[Autodoc] Setup automated documentation for the Scapy codebase

### PR DESCRIPTION
**Pull request by gpotter2**
This PR:
- registers `autodoc` in sphinx to generate the reference API automatically. **THIS:**
  - Will automatically check that the docstring are correctly built (whether Numpy/Google/restructed text style)
  - Builds the reference in the docs
- adds a `tox -e generate_docs` util used to regenerate the reference API tree.
- fixes (many) doc issues to make the tests pass.
- removes `NewDefaultValues`

## Do NOT review the second commit: those are automatically generated files. We need to include them because ReadTheDocs won't generate them.

### CODE REVIEW

**Comment by gpotter2** [comment]
Hi @polybassa & sorry for bothering.

This PR is aiming at building automatically the API reference of Scapy, using `autodoc`. This will be an automated process done by ReadTheDocs, which we don't control. In the current state, `autodoc` can't import a few modules, because they require external dependencies / are platform-specific.

I've taken care of the `arch/` but it would be great if you could make it so that `cansocket_*` may be imported (even though they aren't usable), by `autodoc`. If it's not possible, I'll just disable the doc generation for those files.

Thanks!

**Comment by polybassa** [comment]
Hi,
That’s an awesome PR and I really would like to support you.

Unfortunately I don’t have time in the next two weeks. If that’s to late, you can exclude the cansocket stuff and maybe assign me an issue.

**Comment by gpotter2** [comment]
Thanks ! No worries (enjoy your holidays if any :smile:)

**Comment by polybassa** [comment]
:+1:

File: scapy/arch/bpf/__init__.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Guillaume Valadon <guillaume@valadon.net>
 
-"""
-Scapy *BSD native support
+r"""
+Scapy \*BSD native support
```
Comment (p-l-) [inline scapy/arch/bpf/__init__.py]: What is this?

File: scapy/arch/bpf/consts.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # Guillaume Valadon <guillaume@valadon.net>
 
-"""
-Scapy *BSD native support - constants
+r"""
+Scapy \*BSD native support - constants
```
Comment (p-l-) [inline scapy/arch/bpf/consts.py]: same

File: scapy/arch/bpf/__init__.py (hunk 1 above)
Comment (gpotter2) [reply scapy/arch/bpf/__init__.py]: This is supposed to be valid ReST. A `*` marks the beginning of bold text :/

Comment (p-l-) [reply scapy/arch/bpf/__init__.py]: OK... Maybe "Scapy BSD native support" would work?

File: scapy/arch/bpf/__init__.py [hunk 3]
```diff
@@ -1,5 +1,5 @@
 # Guillaume Valadon <guillaume@valadon.net>
 
-"""
-Scapy *BSD native support
+r"""
```
Comment (gpotter2) [inline scapy/arch/bpf/__init__.py]:
```suggestion
"""
```

File: scapy/arch/bpf/consts.py [hunk 4]
```diff
@@ -1,7 +1,7 @@
 # Guillaume Valadon <guillaume@valadon.net>
 
-"""
-Scapy *BSD native support - constants
+r"""
```
Comment (gpotter2) [inline scapy/arch/bpf/consts.py]:
```suggestion
"""
```

**Comment by gpotter2** [comment]
@p-l- @guedou Should be good, please review & merge.

I'd like it to impose some documentation standards to future PRs :-)

**Pull request merged**